src/state/reducers/month-state/schedule-data/month-switch.action-creator.ts

```diff
@@ -1,51 +1,68 @@
+
+/* eslint-disable @typescript-eslint/camelcase */
+
+import { ScheduleKey, ThunkFunction } from "../../../../api/persistance-store.model";
+import { fetchOrCreateMonthDM, ScheduleDataActionCreator } from "./schedule-data.action-creator";
 import * as _ from "lodash";
-import { createMonthKey, HistoryReducerActionCreator } from "../../history.reducer";
-import { getDateWithMonthOffset } from "./common-reducers";
+import { HistoryReducerActionCreator } from "../../history.reducer";
+import { copyShiftsToMonth, cropMonthInfoToMonth, getDateWithMonthOffset } from "./common-reducers";
+import {
+  cropScheduleDMToMonthDM,
+  MonthDataModel,
+} from "../../../../common-models/schedule-data.model";
 
 export class MonthSwitchActionCreator {
   static switchToNewMonth(offset: number): ThunkFunction<unknown> {
     return async (dispatch, getState): Promise<void> => {
       const history = getState().history;
-      const actualSchedule = getState().actualState;
-      const year = actualSchedule.persistentSchedule.present.schedule_info.year;
-      const month = actualSchedule.persistentSchedule.present.schedule_info.month_number;
+      const actualSchedule = getState().actualState.persistentSchedule.present;
+      const actualMonth = cropScheduleDMToMonthDM(actualSchedule);
+      const { month, year } = actualMonth.scheduleKey;
 
-      const historyAction = HistoryReducerActionCreator.addToHistory(actualSchedule);
+      const historyAction = HistoryReducerActionCreator.addToMonthHistory(actualMonth);
 
       const newDate = getDateWithMonthOffset(month, year, offset);
-      const nextSchedule = history[createMonthKey(newDate.getMonth(), newDate.getFullYear())];
-      if (nextSchedule) {
-        dispatch(ScheduleDataActionCreator.setSchedule(nextSchedule.persistentSchedule.present));
-        if (!history[createMonthKey(month, year)]) {
-          dispatch(historyAction);
-        }
-        return;
-      }
-
-      const newState = _.cloneDeep(scheduleDataInitialState);
-
-      newState.schedule_info.month_number = newDate.getMonth();
-      newState.schedule_info.year = newDate.getFullYear();
-
-      const newPersistentScheduleAction = ScheduleDataActionCreator.addNewSchedule(
-        "PERSISTENT",
-        newState
-      );
-      const newTemporaryScheduleAction = ScheduleDataActionCreator.addNewSchedule(
-        "TEMPORARY",
-        newState
-      );
+      const newMonthKey = new ScheduleKey(newDate.getMonth(), newDate.getFullYear());
+
+      const nextMonth = await fetchOrCreateMonthDM(newMonthKey, history, actualMonth);
+      const addNewScheduleAction = ScheduleDataActionCreator.setScheduleFromMonthDM(nextMonth);
+
       dispatch(historyAction);
-      dispatch(newPersistentScheduleAction);
-      dispatch(newTemporaryScheduleAction);
+      dispatch(addNewScheduleAction);
     };
   }
+
+  static copyActualMonthToMonthWithOffset(offset: number): ThunkFunction<unknown> {
```

Comment:
Kopiowanie przeniosłem z Schedule Action Creatora do tego miejsca, ponieważ wydawało mi się to logiczniejsze + tam już jest sporo akcji